### PR TITLE
Add delete channel feature

### PR DESCRIPTION
**Pull request by bevzaanton**
# Submit a pull request

## Description of the pull request
- add field `deletedAt` to Channel model. Also, add it to gson adapter.
- add API method to `Client` to delete a channel.
- add method to `Channel` to delete this channel.
- add method to `ChannelListViewModel` to delete channel. _I am not sure this method is needed. Client can call Channel.delete or Client.deleteChannel directly._
- add handling of the event `channel.deleted`. 
- add method to `Storage` to delete the channel from local DB.

### CODE REVIEW

**Comment by bevzaanton** [comment]
channelDecodeTest was failed because there is this line ```assertEquals("Thu Oct 03 13:09:28 CEST 2019", channel.getLastMessageDate().toString());```
For me it will be always fail because I live in another time zone.

File: library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java [hunk 1]
```diff
@@ -168,7 +167,11 @@ public void onChannelUpdated(Channel channel, Event event) {
                     channel.handleChannelUpdated(channel);
                 }
 
-                // TODO: what about deleted channels?
+                @Override
+                public void onChannelDeleted(Channel channel, Event event) {
+                    storage().deleteChannel(channel);
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: we should also remove the channel from `client.activeChannels`

File: library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java [hunk 2]
```diff
@@ -214,6 +216,31 @@ public void onError(String errMsg, int errCode) {
         });
     }
 
+    /**
+     * removes the channel. Messages are permanently removed. //TODO I am not sure this method is needed. Client can call Channel.delete or Client.deleteChannel directly.
+     *
+     * @param channel  the channel needs to delete
+     * @param callback the result callback
+     */
+    public void deleteChannel(@NotNull Channel channel, @Nullable ResultCallback<Channel, String> callback) {
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelListViewModel.java]: I think a better approach is to remove this method from the view model and instead let the developer user `client.deleteChannel`. The view model then can register to `channel.deleted` events to update the list of channels

File: library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java (hunk 1 above)
Comment (bevzaanton) [reply library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: There is a strange logic of handling WS events. 
```java
@Override
    public void onWSEvent(Event event) {
        builtinHandler.dispatchEvent(this, event);

        for (int i = eventSubscribers.size() - 1; i >= 0; i--) {
            ChatEventHandler handler = eventSubscribers.get(i);
            handler.dispatchEvent(this, event);
        }

        Channel channel = getChannelByCid(event.getCid());
        if (channel != null) {
            channel.handleChannelEvent(event);
        }
    }
```
Channel model contains link on event subscriber. So if I remove channel from `client.activeChannels` viewModel will not get callback on this event.

Comment (bevzaanton) [reply library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: If I move calling of `buildinHandler` after `channel.handleChannelEvent` it works well. But I don't know maybe there is some logic that I will break is I do it.

**Pull request merged**